updater/lib/dependabot/updater/dependency_group_change_batch.rb

```diff
@@ -14,7 +14,7 @@ def initialize(initial_dependency_files:)
         @updated_dependencies = []
 
         @dependency_file_batch = initial_dependency_files.each_with_object({}) do |file, hsh|
-          hsh[file.path] = { file: file, changed: false, changes: 0 }
+          hsh[file.path] = { file: file, updated_dependencies: [], changed: false, changes: 0 }
```

Comment:
Noting for the future: This seems like primitive obsession, and I think we'd be better off creating an object to handle these operations. I'd expect that to be easier to work with and less prone to making mistakes 🙂

Reply:
Thanks for noting that! I think that would be appropriate for the follow-up work of when we switch the `DependencyGroupChangeBatch` over to read from the new `updated_dependency_file[updated_dependencies]` field